### PR TITLE
Operator Fixup

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
While migrating stuff from Helm to Go, I made some typos and let some things "broken on purpose", mainly because testing one controller-manager without the rest is near impossible.

Now that all resources are reconciled via the Operator, it was time to fix these typos. That's this PR, a random collection of things neither me nor my reviewers noticed ^^

* To make it more obvious and user-friendly which version is going to be deployed, I added a nice helpful welcome message when the operator boots up and also added new `...DOCKERTAG` variables, because the existing GITCOMMIT and GITTAG values were not really suitable to contain Git hashes during development and tags in final releases.
* The PDB for the seed-controller-manager wasn't being reconciled.
* Some resources were not watched, even though they were managed by the operator.
* The rest is basically just typos.

With this code, the operator is able to successfully reconcile master and seed clusters and all components run without errors. :-)

**Which issue(s) this PR fixes**:
Relates to #4723 

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: api/hack/ci/ci-push-images.sh [hunk 1]
```diff
@@ -16,10 +17,21 @@ docker ps > /dev/null 2>&1 || start-docker.sh
...
-# or the name of the base branch in case of a PR.
-DASHBOARDCOMMIT="$(get_latest_dashboard_hash "${PULL_BASE_REF}")"
+# prepare special variables that will be injected into the Kubermatic Operator;
+# use the latest tagged version of the dashboard when we ourselves are a tagged
+# release
+KUBERMATICDOCKERTAG="${GIT_HEAD_TAG:-$GIT_HEAD_HASH}"
+UIDOCKERTAG="$KUBERMATICDOCKERTAG"
+
+if [ -z "$GIT_HEAD_TAG" ]; then
+  UIDOCKERTAG="$(get_latest_dashboard_hash "${PULL_BASE_REF}")"
```
Comment (kron4eg) [inline api/hack/ci/ci-push-images.sh:27]: * how `UIDOCKERTAG` variable is used?
* where does `$PULL_BASE_REF` comes from (seems undeclared) and will error out.

Comment (xrstf) [reply api/hack/ci/ci-push-images.sh:27]: * `UIDOCKERTAG` env variable is used by the `api/Makefile` to inject the version into the source. Just like `GIT_HEAD_HASH` will be magically re-used by the Makefile.
* `PULL_BASE_REF` is defined by Prow when running the CI script.

Comment (xrstf) [reply api/hack/ci/ci-push-images.sh:27]: > go build -v -ldflags '-w -extldflags '-static' -X github.com/kubermatic/kubermatic/api/pkg/resources.KUBERMATICCOMMIT=29ec7bbaa3d178376a8fe4eb314094f757c400eb -X github.com/kubermatic/kubermatic/api/pkg/resources.KUBERMATICGITTAG=weekly-2019-46-64-g29ec7bbaa -X github.com/kubermatic/kubermatic/api/pkg/controller/operator/common.KUBERMATICDOCKERTAG=29ec7bbaa3d178376a8fe4eb314094f757c400eb -X github.com/kubermatic/kubermatic/api/pkg/controller/operator/common.UIDOCKERTAG=NA' -o _build/userdata-openshift ./cmd/userdata-openshift

Doesn't seem to work though, weirdly enough.

Comment (kron4eg) [reply api/hack/ci/ci-push-images.sh:27]: But `UIDOCKERTAG` is not exported so make wouldn't see it

Comment (xrstf) [reply api/hack/ci/ci-push-images.sh:27]: `KUBERMATICDOCKERTAG` is also not exported, yet it works.

Comment (xrstf) [reply api/hack/ci/ci-push-images.sh:27]: Ah brainfart, the ci-push-images is only used in the deploy jobs.

Comment (xrstf) [reply api/hack/ci/ci-push-images.sh:27]: So if it makes you feel better, I can export the variable, but it's not required to do so.

Comment (xrstf) [reply api/hack/ci/ci-push-images.sh:27]: I've exported them now.

**Review by kron4eg — APPROVED** [review]
/lgtm
/approve

**Pull request merged**